engine/jobmaster/dm/dm_jobmaster.go

```diff
@@ -93,23 +92,19 @@ func (j dmJobMasterFactory) NewWorkerImpl(dCtx *dcontext.Context, workerID frame
 	return jm, nil
 }
 
+// initComponents initializes components of dm job master
+// it need to be called firstly in InitImpl and OnMasterRecovered
 func (jm *JobMaster) initComponents(ctx context.Context) error {
 	jm.Logger().Info("initializing the dm jobmaster components")
-	jm.messageAgent = dmpkg.NewMessageAgent(jm.ID(), jm, jm.messageHandlerManager, jm.Logger())
-	if err := jm.messageAgent.Init(ctx); err != nil {
-		return err
-	}
 	taskStatus, workerStatus, err := jm.getInitStatus()
-	if err != nil {
-		return err
-	}
-
-	jm.checkpointAgent = checkpoint.NewCheckpointAgent(jm.initJobCfg, jm.Logger())
+	// still create components if any error
+	// CloseImpl/StopImpl will be called later to close components
```

Comment:
This behaviour should be put in a more obvious place.

Reply:
address in [1d8a473](https://github.com/pingcap/tiflow/pull/6706/commits/1d8a47383c8c88266b291f4ff9690ed3d4d25df1)